tests/test_fakes.py

```diff
@@ -707,6 +707,15 @@ def test_get_path_precedence(cur: snowflake.connector.cursor.SnowflakeCursor):
     cur.execute("select {'K1': {'K2': 1}} as col where col:K1:K2 > 0")
     assert indent(cur.fetchall()) == [('{\n  "K1": {\n    "K2": 1\n  }\n}',)]
 
+    cur.execute("select {'K1': {'K2': 1}} as col where col:K1:K2 > 0")
+    assert indent(cur.fetchall()) == [('{\n  "K1": {\n    "K2": 1\n  }\n}',)]
+
+    cur.execute("create table example (col variant)")
+
+    cur.execute("""insert into example values ('{"K1": "a", "K2": "b"}')""")
```

Comment:
This gives me the following on the real Snowflake:
> ProgrammingError: 002023 (22000): SQL compilation error:
Expression type does not match column data type, expecting VARIANT but got VARCHAR(22) for column COL

Reply:
Weirdly I've made mistakes moving queries from Snowflake worksheet to tests. 

What is your strategy on testing against real Snowflake?

Reply:
Btw, tests fails due to sqlalchemty, guessing something to do with `snowflake-sqlalchemy` ?

Reply:
Not sure which versions get installed on cached venv previously, but on my local machine it was;
```
snowflake-sqlalchemy==1.5.1
SQLAlchemy==1.4.52
```

on failing run;

```
snowflake-sqlalchemy==1.5.2
SQLAlchemy==2.0.29
```

Reply:
> What is your strategy on testing against real Snowflake?

I copy-paste the test into [snowflake.ipynb](https://github.com/tekumara/fakesnow/blob/main/notebooks/snowflake.ipynb) and run it there.

Reply:
Looks like snowflake-sqlalchemy unpinned SQLAlchemy but its now fixed in [1.5.3](https://github.com/snowflakedb/snowflake-sqlalchemy/releases/tag/v1.5.3). So I reran the GHA workflow and its now passed.